File: LENGUAJES/JAVASCRIPT/Ejercicio12.js
//Tabla de la verdad

function generarCombinaciones(variables) {
    // Genera todas las combinaciones de valores booleanos para las variables dadas.
    const combinaciones = [];
    const totalCombinaciones = Math.pow(2, variables.length);

    for (let i = 0; i < totalCombinaciones; i++) {
        const combinacion = [];
        for (let j = 0; j < variables.length; j++) {
            combinacion.push(Boolean((i >> (variables.length - 1 - j)) & 1));
        }
        combinaciones.push(combinacion);
    }

    return combinaciones;
}

function imprimirTablaVerdad(variables) {
    // Imprime la tabla de verdad para la operación A y B.
    const combinaciones = generarCombinaciones(variables);
    
    // Imprimimos la cabecera de la tabla
    console.log(`${variables.join(' | ')} | ${variables.join(' y ')}`);
    console.log('-'.repeat(variables.length * 4 + 10));  // Ajusta el ancho de la línea

    // Evaluamos la expresión para cada combinación
    for (const combinacion of combinaciones) {
        const resultado = combinacion.every(Boolean);  // A y B
        console.log(combinacion.map(String).join(' | ') + ` | ${resultado}`);
    }
}

function main() {
    // Definimos las variables
    const variables = ['A', 'B'];
    imprimirTablaVerdad(variables);
}
main();

File: LENGUAJES/PYTHON/Ejercicio12.py
#TABLA DE LA VERDAD

import itertools

def generar_combinaciones(variables):
    #Genera todas las combinaciones de valores booleanos para las variables dadas.
    return list(itertools.product([True, False], repeat=len(variables)))

def imprimir_tabla_verdad(variables):
    #Imprime la tabla de verdad para la operación A y B.
    combinaciones = generar_combinaciones(variables)
    
    # Imprimimos la cabecera de la tabla
    print(f"{' | '.join(variables)} | {' y '.join(variables)}")
    print("-" * (len(variables) * 4 + 10))  # Ajusta el ancho de la línea

    # Evaluamos la expresión para cada combinación
    for combinacion in combinaciones:
        resultado = all(combinacion)  # A y B
        print(" | ".join(map(str, combinacion)) + f" | {resultado}")

def main():
    # Definimos las variables
    variables = ['A', 'B']
    imprimir_tabla_verdad(variables)

# Llamar a la función principal
if __name__ == "__main__":
    main()